### PR TITLE
core(cli): accept flags from path

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
In order to easily accept an LH settings object over in Lighthouse CI, we need a way to easily passthrough the object. This can be done with config today but you miss out on some nice features like being able to use `--budgets-path`, set the `--output-path`, etc. 

This is a low-cost way to enable consumers of the LH CLI that run in a child process programmatic access to all our CLI flags without manually (and error-pronely) constructing an argument list.

### CODE REVIEW

File: lighthouse-cli/cli-flags.js [hunk 1]
```diff
@@ -62,6 +62,9 @@ function getFlags(manualArgv) {
           'lighthouse <url> --only-categories=performance,pwa',
           'Only run specific categories.')
 
+      // Accept a file for all of these flags.
+      .config('cli-settings-path')
```
Comment (brendankenny) [inline lighthouse-cli/cli-flags.js]: can you add a description specifying that other flags will override this one? (e.g. `--throttling-method=provided` on the command line will override `"throttling-method": "devtools"` in the settings file)

Comment (brendankenny) [reply lighthouse-cli/cli-flags.js]: also, bikeshed time: should we call it something like `cli-flags-path`?

Comment (brendankenny) [reply lighthouse-cli/cli-flags.js]: also, copy over to readme :)

Comment (patrickhulce) [reply lighthouse-cli/cli-flags.js]: > can you add a description specifying that other flags will override this one?

Yes, test added too! 👍 

> cli-flags-path

WFM!

> also, copy over to readme 

This feels close to an autopopulate script too if only we made changes more frequently 🤔

**Comment by paulirish** [comment]
wfm! 👍

File: lighthouse-cli/cli-flags.js (hunk 1 above)
Comment (brendankenny) [reply lighthouse-cli/cli-flags.js]: > This feels close to an autopopulate script too if only we made changes more frequently 🤔

definitely, and it's a little silly we have it at all, but I know I've at least found it useful in the past, so....

**Comment by patrickhulce** [comment]
> I don't entirely understand how you're going to use it

Basically I want to use the existing logic for `outputPath` so I can't use the node integration, but I also want to pass a budget directly as JSON, which I can't really do from CLI flags :)

**Pull request merged**